### PR TITLE
Calculate codon counts

### PR DESCRIPTION
**Pull request by evfaust**
- created gene_features.py file
- defined Gene class
- defined calculate_codon_counts (Gene method)
- created test_Gene.py file
- wrote pytests for Gene class initialization and calculate_codon_counts method

### CODE REVIEW

File: src/vhip/mlmodel/gene_features.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        """Calculate counts of each unique codon in a gene.
+
+        Return:
+            codon_dict (str: int): Each key of dictionary is a unique codon, and the values represent the number of times the associated codon (key) appears in the provided gene sequence.
+            number_imprecise_codons (int): Number of codons that are not precise (i.e. are not found in expected CODON_LIST).
+        """
+        self.number_imprecise_codons = 0
+        self.codon_dict = dict.fromkeys(CODON_LIST, 0)
+
+        for i in range(0, len(self.seq), self.codon_length):
```
Comment (GBastion) [inline src/vhip/mlmodel/gene_features.py:106]: What happens if the length of the gene sequence is not divisible by the codon_length?

File: tests/test_Gene.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+from vhip.mlmodel.gene_features import Gene
+
+
+def test_Gene_attributes():
+    """Test code to create Gene object and initialize class attributes."""
+    test_gene = Gene("ATGCCGATT")
+    assert test_gene.seq == "ATGCCGATT"
+    assert test_gene.codon_length == 3
+
```
Comment (GBastion) [inline tests/test_Gene.py:11]: Add a test for the edge case where gene length is not divisible by the codon_length

**Review by GBastion — CHANGES_REQUESTED** [review]

File: src/vhip/mlmodel/gene_features.py (hunk 1 above)
Comment (evfaust) [reply src/vhip/mlmodel/gene_features.py:109]: I added a condition that prevents Gene initialization and raises an Exception if the sequence length is not a multiple of codon length. When extracting info from GeneSets later on, I can write code to handle these excpetions. I think it is most conservative to skip these genes because they may be partial genes and could add noise to predictions.

File: tests/test_Gene.py (hunk 2 above)
Comment (evfaust) [reply tests/test_Gene.py:17]: Got it. Added check for if an Excpetion gets raised when the gene length is not divisible by codon length.

**Review by GBastion — APPROVED** [review]
Looks good!
